angular/src/app/shared/components/table/table.component.ts

```diff
@@ -0,0 +1,144 @@
+import {
+  Component,
+  ChangeDetectionStrategy,
+  ViewChild,
+  AfterViewInit,
+  ChangeDetectorRef,
+  ElementRef,
+} from '@angular/core';
+
+import { MatPaginator } from '@angular/material/paginator';
+import { MatSort } from '@angular/material/sort';
+
+import { debounceTime, distinctUntilChanged, fromEvent, map, merge, startWith, switchMap, tap } from 'rxjs';
+
+import { Database } from '../../../core/services/database.service';
+import { FilmModel } from '../../../core/models/film.model';
+
+/**
+ * Table component.
+ */
+@Component({
+  selector: 'sw-table',
+  templateUrl: './table.component.html',
+  styleUrls: ['./table.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class TableComponent implements AfterViewInit {
+
+  public constructor(private database: Database, private changeDetectorRefs: ChangeDetectorRef) {
+  }
+
+  /** Columns to show. */
+  public displayedColumns: string[] = ['title', 'director', 'producer', 'created'];
+
+  /** Data to fill table.*/
+  public data!: FilmModel[];
+
+  /** Len of data.*/
+  public resultsLength = 0;
+
+  /** Flag is data loading.*/
+  public isLoadingResults = true;
+
+  /** Flag is rate limit is reached.*/
+  public isRateLimitReached = false;
+
+  /** Elements on page.*/
+  public pageSize = 2;
+
+  /** Index of previous page.*/
+  public previousPage!: number;
+
+  /** Filter input value.*/
+  public filterInputValue = '';
+
+  /**
+   * @inheritDoc
+   * Reference on Paginator.
+   */
+  @ViewChild(MatPaginator, { static: false }) private paginator!: MatPaginator;
+
+  /**
+   * @inheritDoc
+   * Reference on sort table.
+   */
+  @ViewChild(MatSort) private sort!: MatSort;
```

Comment:
We prefer to write decorator and property on different lines.

```suggestion
  @ViewChild(MatSort) 
  private sort!: MatSort;
```